cmake/kokkos_functions.cmake

```diff
@@ -660,6 +660,13 @@ MACRO(kokkos_find_imported NAME)
 
   MARK_AS_ADVANCED(${NAME}_INCLUDE_DIRS ${NAME}_FOUND_LIBRARIES ${NAME}_LIBRARY)
 
+  #this is so much fun on a Cray system
+  #/usr/include should never be added as a -isystem include
+  #this freaks out the compiler include search order
+  IF (KOKKOS_IS_CRAYPE)
+    LIST(REMOVE_ITEM ${NAME}_INCLUDE_DIRS "/usr/include")
```

Comment:
This looks a bit scary.

Reply:
Not as scary as getting a `<stdlib.h> cannot be found error` because `/usr/include` has been removed from your compiler search path via a cray reordering of the search path.  I'm open to other ways to fix this.

Reply:
I was not requesting to change this. If this is what it takes to make `Cray` compilers happy, so be it.

Reply:
You are right, though. It IS scary.